### PR TITLE
doc/hooks/xfce4-dev-tools: document hook

### PR DESCRIPTION
**Pull request by tomodachi94**
Part of #341479

Related: https://github.com/orgs/NixOS/projects/67/views/1?pane=issue&itemId=84912848

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tomodachi94** [comment]
Also going to request a review from @bobby285271, who has done some work on this package in particular.

File: doc/hooks/xfce4-dev-tools.section.md
Comment (bobby285271) [inline doc/hooks/xfce4-dev-tools.section.md:5]: https://github.com/NixOS/nixpkgs/blob/c77dd94ccb8b3873505ef3d98dc9d3c6571f2d05/pkgs/desktops/xfce/core/xfce4-dev-tools/setup-hook.sh#L1-L5

Heh, I doubt if "envHooks" actually works [these days](https://github.com/NixOS/nixpkgs/commit/4d2b7638173afb4c0f0c12a43654bcff5c9500ab). Though in any case I think I will try removing these lines [soon](https://github.com/bobby285271/nixpkgs/blob/upd/xfce-4.20/pkgs/desktops/xfce/core/xfce4-dev-tools/setup-hook.sh) :joy:

Comment (tomodachi94) [reply doc/hooks/xfce4-dev-tools.section.md:5]: > Though in any case I think I will try removing these lines [soon](https://github.com/bobby285271/nixpkgs/blob/upd/xfce-4.20/pkgs/desktops/xfce/core/xfce4-dev-tools/setup-hook.sh) 😂

Should we wait until then to merge this PR?

Comment (tomodachi94) [reply doc/hooks/xfce4-dev-tools.section.md]: Removed this part of the manual section, since it doesn't work and will be removed by https://github.com/NixOS/nixpkgs/pull/351781

File: doc/hooks/xfce4-dev-tools.section.md
Comment (bobby285271) [inline doc/hooks/xfce4-dev-tools.section.md:5]: For some reason `{#dontUseXdtAutogenPhase}` is rendered as-is for me though I also see this elsewhere, searching `{#` :joy:

**Review by bobby285271 — COMMENTED** [review]
The content looks correct for me, so the content LGTM :joy:

File: doc/hooks/xfce4-dev-tools.section.md
Comment (tomodachi94) [reply doc/hooks/xfce4-dev-tools.section.md]: Fixed. Apparently, this needs to be a pseudo-link in the form `[]{#dontUseXdtAutogenPhase}` when it's not part of a heading.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**
